eclair-core/src/test/scala/fr/acinq/eclair/api/ApiServiceSpec.scala

```diff
@@ -0,0 +1,277 @@
+        blockHeight = 9999,
+        publicAddresses = NodeAddress.fromParts("localhost", 9731).get :: Nil
+      ))
+    })
+
+    Post("/getinfo") ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val resp = entityAs[String]
+        assert(resp.toString.contains(Alice.nodeParams.nodeId.toString))
+        matchTestJson("getinfo", false, resp)
+      }
+  }
+
+  test("'close' method should accept a shortChannelId") {
+
+    val shortChannelIdSerialized = "42000x27x3"
+
+    val mockService = new MockService(new EclairMock {
+      override def close(channelIdentifier: Either[ByteVector32, ShortChannelId], scriptPubKey: Option[ByteVector]): Future[String] = {
+        Future.successful(Alice.nodeParams.nodeId.toString())
+      }
+    })
+
+    Post("/close", FormData("shortChannelId" -> shortChannelIdSerialized).toEntity) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      addHeader("Content-Type", "application/json") ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        val resp = entityAs[String]
+        assert(resp.contains(Alice.nodeParams.nodeId.toString))
+        matchTestJson("close", false, resp)
+      }
+  }
+
+  test("'connect' method should accept an URI and a triple with nodeId/host/port") {
+
+    val remoteNodeId = "030bb6a5e0c6b203c7e2180fb78c7ba4bdce46126761d8201b91ddac089cdecc87"
+    val remoteHost = "93.137.102.239"
+    val remotePort = "9735"
+    val remoteUri = "030bb6a5e0c6b203c7e2180fb78c7ba4bdce46126761d8201b91ddac089cdecc87@93.137.102.239:9735"
+
+    val mockService = new MockService(new EclairMock {
+      override def connect(uri: String): Future[String] = Future.successful("connected")
+    })
+
+    Post("/connect", FormData("nodeId" -> remoteNodeId, "host" -> remoteHost, "port" -> remotePort).toEntity) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        assert(entityAs[String] == "\"connected\"")
+      }
+
+    Post("/connect", FormData("uri" -> remoteUri).toEntity) ~>
+      addCredentials(BasicHttpCredentials("", mockService.password)) ~>
+      Route.seal(mockService.route) ~>
+      check {
+        assert(handled)
+        assert(status == OK)
+        println(entityAs[String])
+        assert(entityAs[String] == "\"connected\"")
+      }
+  }
+
+  private def matchTestJson(apiName: String, overWrite: Boolean, response: String) = {
+    val p = Paths.get(s"src/test/resources/api/$apiName")
+
+    if (overWrite) {
+      Files.writeString(p, response)
+      assert(false, "'overWrite' should be false before commit")
+    } else {
+      val expectedResponse = Source.fromFile(p.toUri).mkString
```

Comment:
Accessing resources should not be done through the file API but with `getClass.getResourceAsStream` or `getClass.getResource`, you could try this instead:

``` scala
    val name = s"/api/$apiName"
    val stream = getClass.getResourceAsStream(name)
    val expectedResponse = Try(Source.fromInputStream(stream).mkString).getOrElse(throw new RuntimeException(s"resource /api/$apiName not found, it should contain $response"))
    assert(response == expectedResponse, s"Test mock for $apiName did not match the expected response")
```

Reply:
Done in ba63614